File: 백준문제/C언어/곱하기혹은더하기.c
#include<stdio.h>

//숫자로만 이루어진 문자열을 입력으로 받는다
//각숫자 사이에 연산자를 넣어서 최대값을 만든다.
//모든연산은 왼쪽부터 이루어진다.

int main(void)
{
	char str[20];
	scanf_s("%s", str, sizeof(str) / 1);
	int result = str[0] - '0';
	for (int i = 1; str[i] != '\0'; i++)
	{
		int num = str[i] - '0';
		if (result <= 1 || num <= 1)
			result += num;
		else
			result *= num;
	}
	printf("%d", result);
}

File: 백준문제/Python/greedy/곱하기혹은더하기.py
# 숫자로만 이루어진 문자열을 입력으로 받는다
# 각숫자 사이에 연산자를 넣어서 최대값을 만든다.
# 모든연산은 왼쪽부터 이루어진다.

data = input()
result = int(data[0])

for i in range(1, len(data)):
    num = int(data[i])
    if num <= 1 or result <= 1:
        result += num
    else :
        result *= num
print(result)











